# Statistics for Decision Making

## Instructions:

• These problems will test your ability to manipulate real-world data and answer statistical questions.

• Use the property.csv dataset to arrive at the solutions.

• The data is on real estate prices in Australia. Use this data to test the following hypothesis:


----
## 1. For the suburb of Altona, it is postulated that a typical property sells for $800,000.

• Use the data at hand to test this assumption.

• Is the typical property price really $800,000 or has it increased?

• Use a significance level of 5%.

----

In [1]:
import pandas as pd
from scipy import stats

# Load dataset
df = pd.read_csv("property.csv")

# Filter Altona properties
altona = df[df['Suburb'] == 'Altona']['Price']

# One-sample t-test against 800,000
t_stat, p_val = stats.ttest_1samp(altona, 800000)

print("Sample size:", len(altona))
print("Mean price:", altona.mean())
print("t-statistic:", t_stat)
print("p-value (one-tailed):", p_val/2)


Sample size: 74
Mean price: 834830.4054054054
t-statistic: 1.0277020770199676
p-value (one-tailed): 0.1537416356527775


**Decision:**
----
**H₀ (Null)**: The average property price in Altona is $800,000.

**H₁ (Alternative)**: The average property price in Altona is greater than $800,000.

This is a **one-tailed one-sample t-test**.

**Interpretation:**
----
Since **p = 0.154 > 0.05**, we **fail to reject H₀**.

This means the evidence is **not strong enough** to conclude that Altona’s average property price is significantly higher than $800,000.

----
## 2. For the year 2016, is there any difference in the prices of properties sold in the summer months vs winter months?

• Consider months from October till March as winter months and rest as summer months.

• Use a significance level of 5%.

----

In [2]:
import pandas as pd
from scipy import stats

# Load dataset
df = pd.read_csv("property.csv")

# Parse dates with dayfirst=True
df['Date'] = pd.to_datetime(df['Date'], dayfirst=True)

# Filter for year 2016
df_2016 = df[df['Date'].dt.year == 2016]

# Define seasons
winter = df_2016[df_2016['Date'].dt.month.isin([10,11,12,1,2,3])]['Price']
summer = df_2016[df_2016['Date'].dt.month.isin([4,5,6,7,8,9])]['Price']

# Independent two-sample t-test
t_stat, p_val = stats.ttest_ind(winter, summer, equal_var=False)

print("Winter mean:", winter.mean())
print("Summer mean:", summer.mean())
print("t-statistic:", t_stat)
print("p-value:", p_val)


Winter mean: 1116647.5917391304
Summer mean: 1048054.7286917741
t-statistic: 3.9211012778003753
p-value: 8.950129747463378e-05


**Decision:**
----
**H₀ (Null)**: Average property prices in summer = average property prices in winter (no seasonal difference).

**H₁ (Alternative)**: Average property prices in summer ≠ average property prices in winter (seasonal difference exists).

This is a **two-tailed independent samples t-test**.

**Interpretation:**
----
Since **p-value = 0.0000895 < 0.05** , we **reject the null hypothesis**.

This confirms that **property prices in 2016 were significantly higher in winter than in summer**.

----
## 3. For the suburb of Abbotsford, what is the probability that out of 10 properties sold, 3 will not have a car parking space?

• Use the column car in the dataset.

• Round off your answer to 3 decimal places.

----

In [3]:
import pandas as pd
from scipy.stats import binom

df = pd.read_csv("property.csv")

# Filter Abbotsford properties
abbotsford = df[df['Suburb'] == 'Abbotsford']

# Probability of no car space
p_no_car = (abbotsford['Car'] == 0).mean()
print("Probability of no car space:", p_no_car)

# Binomial probability: exactly 3 out of 10
prob = binom.pmf(3, 10, p_no_car)
print("Final probability:", round(prob, 3))


Probability of no car space: 0.26785714285714285
Final probability: 0.26


**Decision:**
----
It’s called a **Binomial Probability Calculation** (or **Binomial Distribution model**).

Probability = 0.26

**Interpretation:**
----
About **26% chance** that **exactly 3 out of 10 Abbotsford properties will lack car parking space**.

----
## 4. In the suburb of Abbotsford, what are the chances of finding a property with 3 rooms? Round your answer to 3 decimal places.
----

In [4]:
import pandas as pd

df = pd.read_csv("property.csv")

# Filter Abbotsford properties
abbotsford = df[df['Suburb'] == 'Abbotsford']

# Probability of exactly 3 rooms
p_3rooms = (abbotsford['Rooms'] == 3).mean()
print("Probability of 3 rooms:", round(p_3rooms, 3))


Probability of 3 rooms: 0.357


**Decision**
----
It’s called a **simple probability (proportion)**.

About **35.7%** of Abbotsford properties in your dataset have exactly 3 rooms.

**Interpretation:**
----
Roughly **1 in 3 properties** in Abbotsford are 3-room houses.

This suggests that 3-room properties are a **common segment of the market**.

----
## 5. In the suburb of Abbotsford, what are the chances of finding a property with 2 bathrooms? Round your answer to 3 decimal places.
----

In [5]:
import pandas as pd

df = pd.read_csv("property.csv")

# Filter Abbotsford properties
abbotsford = df[df['Suburb'] == 'Abbotsford']

# Probability of exactly 2 bathrooms
p_2bath = (abbotsford['Bathroom'] == 2).mean()
print("Probability of 2 bathrooms:", round(p_2bath, 3))


Probability of 2 bathrooms: 0.339


**Decision**
----
This is a **proportion calculation (empirical probability)**.

About **33.9%** of Abbotsford properties in your dataset have exactly 2 bathrooms.

**Interpretation:**
----
Roughly **1 in 3 properties** in Abbotsford are 2-bathroom houses.

This makes 2 bathrooms a **common feature** in the suburb.

----
## 6. One-Sample Hypothesis Test (Industry Pricing)

A real estate firm claims that the average property price in Richmond is $1,000,000. Using the dataset, test whether the actual average price is significantly different from this claim at a 5% significance level. Clearly state:

• Null and alternative hypotheses

• Test statistic

• p-value

• Final business conclusion

----

In [6]:
import pandas as pd
from scipy import stats

df = pd.read_csv("property.csv")

# Filter Richmond properties
richmond = df[df['Suburb'] == 'Richmond']['Price']

# One-sample t-test against $1,000,000
t_stat, p_val = stats.ttest_1samp(richmond, 1000000)

print("Richmond mean:", richmond.mean())
print("t-statistic:", t_stat)
print("p-value:", p_val)


Richmond mean: 1083564.423076923
t-statistic: 2.579547704074923
p-value: 0.01044499066415202


**Decision:**
----
**H₀ (Null Hypothesis)**: The average property price in Richmond = $1,000,000.

**H₁ (Alternative Hypothesis)**: The average property price in Richmond ≠ $1,000,000.

This is a **two-tailed one-sample t-test**.

**Interpretation:**
----
Since **p-value = 0.0104 < 0.05**, we **reject H₀**.

Statistically, Richmond’s average property price is **significantly different** from $1,000,000

The actual average price in Richmond is **higher (~$1.08M)** than the claimed $1M.

----
## 7. Independent Two-Sample T-Test (Feature Impact)

Do properties with car parking sell at a higher average price than properties without car parking, across the entire dataset? Use a 5% significance level and justify:

• Choice of test

• Interpretation of p-value

• Business implications for developers

----

In [7]:
import pandas as pd
from scipy import stats

df = pd.read_csv("property.csv")

# Split dataset by car parking availability
with_car = df[df['Car'] > 0]['Price']
without_car = df[df['Car'] == 0]['Price']

# Independent two-sample t-test (Welch’s)
t_stat, p_val = stats.ttest_ind(with_car, without_car, equal_var=False)

print("Mean price (with car):", with_car.mean())
print("Mean price (without car):", without_car.mean())
print("t-statistic:", t_stat)
print("p-value:", p_val)


Mean price (with car): 1074443.9241114312
Mean price (without car): 1079088.0107212476
t-statistic: -0.27224703025972535
p-value: 0.7854749985398599


**Decision:**
----
**H₀ (Null Hypothesis)**: μ_with = μ_without (average prices are equal)

**H₁ (Alternative Hypothesis)**: μ_with ≠ μ_without (average prices differ)

This is a **independent two-sample t-test (Welch’s t-test)**.

**Interpretation:**
----
Since **p-value = 0.785 > 0.05**, we **fail to reject H₀**.

This means there is **no statistically significant difference** in average property prices between homes with car parking and those without.

Car parking could still matter for **buyer convenience and marketability**, but it’s not a proven price driver statistically.

----
## 8. Two-Way ANOVA (Location & Property Type)

Investigate whether property prices are influenced by:

• Suburb

• Type of property

• Interaction between suburb and property type

Use Two-Way ANOVA and explain which factors significantly affect price.

----

In [8]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols

# Load dataset
df = pd.read_csv("property.csv")

# Optional: restrict to top suburbs and property types to avoid sparse categories
top_suburbs = df['Suburb'].value_counts().nlargest(5).index
top_types = df['Type'].value_counts().nlargest(3).index
df_sub = df[df['Suburb'].isin(top_suburbs) & df['Type'].isin(top_types)]

# Build Two-Way ANOVA model
model = ols('Price ~ C(Suburb) * C(Type)', data=df_sub).fit()

# Run Type III ANOVA (handles unbalanced designs better)
anova_table = sm.stats.anova_lm(model, typ=3)

print(anova_table)


                         sum_sq      df            F        PR(>F)
Intercept          2.405130e+14     1.0  3217.115237  0.000000e+00
C(Suburb)          4.140875e+13     4.0   138.471457  8.547687e-99
C(Type)            8.052570e+12     2.0    53.855815  3.311226e-23
C(Suburb):C(Type)  6.129412e+12     8.0    10.248420  5.587511e-14
Residual           9.823523e+13  1314.0          NaN           NaN


**Decision:**
----
>**H₀ (Suburb):** Average property prices are equal across suburbs.

>**H₁ (Suburb):** Average property prices differ across suburbs.

>**H₀ (Type):** Average property prices are equal across property types.

>**H₁ (Type):** Average property prices differ across property types.

>**H₀ (Interaction):** No interaction between suburb and property type.

>**H₁ (Interaction):** The effect of property type on price depends on suburb.

We used a **Two-Way ANOVA** because we want to test the effect of **two categorical factors** (Suburb and Property Type) on a continuous variable (Price).

**Interpretation:**
----
**Suburb effect:**

* p < 0.05 → Property prices vary significantly across suburbs.

* Location is a major driver of price differences.

**Property Type effect:**

* p < 0.05 → Property prices differ significantly across types (house, unit, townhouse).

* Type is a key determinant of value.

**Interaction effect (Suburb × Type):**

* p < 0.05 → The impact of property type depends on suburb.

* This means developers cannot generalize property type pricing across all suburbs — it must be suburb-specific.

**Location (suburb) and property type both significantly influence property prices.**

----
## 9. p-Value Interpretation (Decision Making)

A hypothesis test comparing prices across two suburbs results in a p-value of 0.032.

Answer:

• What does this p-value indicate?

• Should the null hypothesis be rejected at α = 0.05?

• How should a business stakeholder interpret this result?

----

**What does this p-value indicate?**
    
* A p-value of 0.032 means that if the null hypothesis (no difference in average property prices between the two suburbs) were true, there is only a 3.2% probability of observing a difference as large as the one found in your data purely by chance.

* In plain terms: the difference in prices is unlikely to be random noise.

**Should the null hypothesis be rejected at α = 0.05?**

* Yes. Since p = 0.032 < 0.05, we reject the null hypothesis.

* This provides statistical evidence that the average property prices between the two suburbs are significantly different.

**How should a business stakeholder interpret this result?**
    
* Property prices are not uniform across suburbs — they differ in a statistically meaningful way.

* Implications:

> Developers should adopt location-specific pricing strategies.

> Premium projects can be positioned in higher-priced suburbs, while more affordable developments can be marketed in lower-priced suburbs.

> Investors should treat each suburb as a distinct market segment, rather than assuming one-size-fits-all pricing.

----
## **10. Industry-Style Hypothesis Validation (Policy Decision)**

A housing policy group believes that properties with more than 2 bathrooms command a premium price.
Design and execute a statistical test to validate this claim:

• Identify the correct test

• State hypotheses

• Report p-value

• Give a clear recommendation to policymakers

----

In [9]:
import pandas as pd
from scipy import stats

# Split into groups
bath_le2 = df[df['Bathroom'] <= 2]['Price']
bath_gt2 = df[df['Bathroom'] > 2]['Price']

# Welch’s t-test
t_stat, p_val = stats.ttest_ind(bath_gt2, bath_le2, equal_var=False)

# One-tailed adjustment
if t_stat > 0:
    one_tailed_p = p_val / 2
else:
    one_tailed_p = 1 - (p_val / 2)

print("Mean price (>2 bathrooms):", bath_gt2.mean())
print("Mean price (≤2 bathrooms):", bath_le2.mean())
print("t-stat:", t_stat)
print("one-tailed p-value:", one_tailed_p)


Mean price (>2 bathrooms): 1882824.2
Mean price (≤2 bathrooms): 1007347.9350638977
t-stat: 28.800231483982945
one-tailed p-value: 3.69021417834531e-137


**Identify the Correct Test**
----

**Independent Two-Sample t-test (Welch’s)** comparing mean property prices of:

* Group 1: Properties with **>2 bathrooms**

* Group 2: Properties with **≤2 bathrooms**

**State Hypotheses**
----

**H₀ (Null)**: μ\{>2 bathrooms} = μ\{≤2 bathrooms}
(No premium for properties with more than 2 bathrooms)

**H₁ (Alternative)**: μ\{>2 bathrooms} > μ\{≤2 bathrooms}
(Properties with more than 2 bathrooms command a premium price)

This is a **one-tailed test** because the claim is directional (premium pricing).

**Report p-value**
----

one-tailed **p-value** = 3.69 × 10⁻¹³⁷

**Decision rule:**

* If p < 0.05, reject H₀ → evidence supports the claim.

* If p ≥ 0.05, fail to reject H₀ → no statistical evidence of a premium.

(Example: If p = 3.69 × 10⁻¹³⁷, then at α = 0.05 we reject H₀ and conclude that >2 bathrooms properties command a premium.)

**Recommendation to Policymakers**
----

* If significant (p < 0.05):

> The claim is validated.

> Policymakers can encourage developers to build larger homes with more bathrooms, as they statistically command higher prices.

> Housing policy can support zoning and incentives for premium housing projects.

* If not significant (p ≥ 0.05):

> The claim is not supported.

> Policymakers should reconsider promoting bathroom count as a driver of housing value.

> Focus should shift to other features (location, bedrooms, car spaces) that may have stronger pricing impact.